### PR TITLE
add useMemo to fix infinite request loop in various hooks

### PR DESCRIPTION
**Pull request by denisdimitrov**
The problem was in the `{}` syntax that was passed to `useLiveCollection`. It was creating a new object reference on each re-render and forcing the `useLiveCollection` hook to retrigger. Also the `parseCollectionQueryOptions` was creating a new reference as well, and forcing the `useEffect` to retrigger and subscribe/unsubscribe.

### CODE REVIEW

**Comment by denisdimitrov** [comment]
@goran-peoski-work I was going to remove the `useLive` implementation and replace it with `useFire` but the website just crashes. It worked fine yesterday before I rebased with `redesign-2021` but now it doesn't.

File: src/hooks/fire/useLiveCollection.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export const useLiveCollection = <T extends object, ID extends string = string>(
     hasDeferred,
     constraints,
     filteredConstraints,
-  } = parseCollectionQueryOptions(options);
+  } = useMemo(() => parseCollectionQueryOptions(options), [options]);
```
Comment (goran-peoski-work) [inline src/hooks/fire/useLiveCollection.ts:38]: Interesting, I didn't think this could work considering `options` itself is an object and maybe it's fields would need direct reference.

Maybe it does not change much or anything if only `options` are passed into `useLiveCollection`, but that R&D would be for another PR.

File: src/hooks/spaces/useSpacesByOwner.tsx [hunk 2]
```diff
@@ -35,7 +33,11 @@ export const useSpacesByOwner: UseSpacesByOwner = ({ worldId, userId }) => {
     [worldId, userId]
   );
 
-  const result = useLiveCollection<SpaceWithId>({ path, constraints });
+  const options = useMemo(() => ({ path: PATH.spaces, constraints }), [
```
Comment (goran-peoski-work) [inline src/hooks/spaces/useSpacesByOwner.tsx:36]: 👍 Nice find. I will have to update some hook APIs to make it harder to fall into this trap in the future.

File: src/hooks/user/usePermission.ts [hunk 3]
```diff
@@ -12,13 +12,11 @@ type AdminRole = {
   allowAll: boolean;
   users: string[];
 };
-
```
Comment (goran-peoski-work) [inline src/hooks/user/usePermission.ts:15]: no rename to `useLivePermission`?

**Review by goran-peoski-work — APPROVED** [review]
LGTM

File: src/hooks/user/usePermission.ts (hunk 3 above)
Comment (denisdimitrov) [reply src/hooks/user/usePermission.ts:15]: My bad, I'll add the change.
